jenkins/pipelines/ci/ticdc/dm_ghpr_compatibility_test.groovy

```diff
@@ -0,0 +1,235 @@
+def TIDB_BRANCH = "master"
+def BUILD_NUMBER = "${env.BUILD_NUMBER}"
+
+def PRE_COMMIT = "${ghprbTargetBranch}"
+def MYSQL_ARGS = "--log-bin --binlog-format=ROW --enforce-gtid-consistency=ON --gtid-mode=ON --server-id=1 --default-authentication-plugin=mysql_native_password"
+def TEST_CASE = ""
+def BREAK_COMPATIBILITY = "false"
+
+println "comment body=${ghprbCommentBody}"
+
+// if this PR breaks compatibility
+def m0 = ghprbCommentBody =~ /break_compatibility\s*=\s*([^\s\\]+)(\s|\\|$)/
+if (m0) {
+    BREAK_COMPATIBILITY = "${m0[0][1]}"
+}
+m0 = null
+println "BREAK_COMPATIBILITY=${BREAK_COMPATIBILITY}"
+if (BREAK_COMPATIBILITY == "true") {
+    currentBuild.result = 'SUCCESS'
+    return 0
+}
+
+// parse tidb branch
+def m1 = ghprbCommentBody =~ /tidb\s*=\s*([^\s\\]+)(\s|\\|$)/
+if (m1) {
+    TIDB_BRANCH = "${m1[0][1]}"
+}
+m1 = null
+println "TIDB_BRANCH=${TIDB_BRANCH}"
+
+// parser previous commit
+def m2 = ghprbCommentBody =~ /pre_commit\s*=\s*([^\s\\]+)(\s|\\|$)/
+if (m2) {
+    PRE_COMMIT = "${m2[0][1]}"
+}
+m2 = null
+println "PRE_COMMIT=${PRE_COMMIT}"
+
+// parser test case name
+def m3 = ghprbCommentBody =~ /case\s*=\s*([^\s\\]+)(\s|\\|$)/
+if (m3) {
+    TEST_CASE = "${m3[0][1]}"
+}
+m3 = null
+println "TEST_CASE=${TEST_CASE}"
+
+def boolean isBranchMatched(List<String> branches, String targetBranch) {
+    for (String item : branches) {
+        if (targetBranch.startsWith(item)) {
+            println "targetBranch=${targetBranch} matched in ${branches}"
+            return true
+        }
+    }
+    return false
+}
+
+println "This build use go1.16"
+GO_BUILD_SLAVE = GO1160_BUILD_SLAVE
+GO_TEST_SLAVE = GO1160_TEST_SLAVE
+POD_GO_DOCKER_IMAGE = "hub.pingcap.net/jenkins/centos7_golang-1.16:latest"
+
+println "BUILD_NODE_NAME=${GO_BUILD_SLAVE}"
+println "TEST_NODE_NAME=${GO_TEST_SLAVE}"
+println "POD_GO_DOCKER_IMAGE=${POD_GO_DOCKER_IMAGE}"
+
+
+catchError {
+    stage('Prepare') {
+        node ("${GO_BUILD_SLAVE}") {
+            container("golang") {
+                def ws = pwd()
+                deleteDir()
+                dir("/home/jenkins/agent/git/ticdc") {
+                    if (sh(returnStatus: true, script: '[ -d .git ] && [ -f Makefile ] && git rev-parse --git-dir > /dev/null 2>&1') != 0) {
+                        deleteDir()
+                    }
+                    try {
+                        checkout changelog: false, poll: false, scm: [$class: 'GitSCM', branches: [[name: 'master']], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'PruneStaleBranch'], [$class: 'CleanBeforeCheckout']], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'github-sre-bot-ssh', refspec: '+refs/pull/*:refs/remotes/origin/pr/*', url: 'git@github.com:pingcap/ticdc.git']]]
```

Comment:
git refspec `'+refs/pull/*:refs/remotes/origin/pr/*'` will pull all pr to local, it will lead to checkout code timeout due to limited idc network bandwidth to github. Try use `"+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*"` to pull current pr code only.

example
```shell
def specStr = "+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*"
if (ghprbPullId == null || ghprbPullId == "") {
    specStr = "+refs/heads/*:refs/remotes/origin/*"
}
```